### ISSUE TITLE
Animated gif from raw buffer?

### ISSUE THREAD

**Issue by spennyp** [lovell/sharp#4092]
### What are you trying to achieve?

I wan't to be able to create an animated gif from a raw buffer. The issue is that when I do so I am unable to set pages, or the pageHeight.

My goal is to take an input gif, extract the raw buffer and recolor pixels, then use that to create an output gif.

I tried to do the following:
```typescript
// Input
const inputSharp = await sharp("input.gif", {animated: true});
const originalMetadata = await inputSharp.metadata();

// Extract raw buffer
const { data, info } = await inputSharp.raw().toBuffer({ resolveWithObject: true });
const pixelArray = new Uint8ClampedArray(data.buffer);
for (let i = 0; i < pixelArray.length; i += info.channels) {
    // Recolor pixel array
}

// Try to recreate the gif
const outputSharp = await sharp(pixelArray, {
    raw: {
      width: info.width,
      height: info.height,
      channels: info.channels,
    },
    animated: true,
 }).gif({ force: true });
const outputMeta = await outputSharp.metadata();
```

The inputMetadata is:
```typescript
{
  format: 'gif',
  size: 68917,
  width: 600,
  height: 28800,
  space: 'srgb',
  channels: 4,
  depth: 'uchar',
  isProgressive: false,
  paletteBitDepth: 2,
  pages: 48,
  pageHeight: 600,
  loop: 0,
  delay: [
    80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80,
    80, 80, 80, 80, 80, 80, 80, 80, 80, 80, 80,
    80, 80, 80,  0,  0,  0,  0,  0,  0,  0,  0,
     0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
     0,  0,  0,  0
  ],
  background: { r: 76, g: 105, b: 113 },
  hasProfile: false,
  hasAlpha: true
}
```

The outputMetadata is:
```typescript
{
  format: 'gif',
  size: 42086,
  width: 600,
  height: 28800,
  space: 'srgb',
  channels: 4,
  depth: 'uchar',
  isProgressive: false,
  paletteBitDepth: 1,
  pages: 1,
  loop: 0,
  delay: [ 0 ],
  background: { r: 76, g: 105, b: 113 },
  hasProfile: false,
  hasAlpha: true
}
```

The output image is a toilet roll of the gif frames. Everything looks exactly like I want but I can't get it to export as an animated gi. I seem to be unable to set the number of pages, or the pageHeight. 

I've tried also setting the delay but it doesn't work.

Any suggestions here?
Basically it seems the issue is its not possible to create an animated gif from a raw buffer.

**Comment by spennyp** [comment]
This looks like the same question: https://github.com/lovell/sharp/issues/3236


It seems like this is still not possible?

**Comment by lovell** [comment]
I've re-opened #3236 with additional context to track this.
